### ISSUE TITLE
LDAP on windows not working

### ISSUE THREAD

**Issue by Jojojov** [curl/curl#4261]
### I did this

     >curl "ldap://ldap.forumsys.com"
     curl: (62) LDAP local: No Memory

### I expected the following

     curl "ldap://ldap.forumsys.com"
     DN:
             objectClass: top
             objectClass: OpenLDAProotDSE

### curl/libcurl version

curl 7.65.3 (x86_64-pc-win32) libcurl/7.65.3 OpenSSL/1.1.1c (Schannel) zlib/1.2.11 brotli/1.0.7 WinIDN libssh2/1.9.0 nghttp2/1.39.2
Release-Date: 2019-07-19

### operating system

Both Windows 10 and Windows Server 2016

**Comment by bagder** [comment]
Any chance you can single-step through the code and see what function call that acts up?

**Comment by Jojojov** [comment]
I will try to find some time soon...

**Comment by samsinsane** [comment]
My apologies, I can see the issue is that it's valid for `conn->data->state.up.query` to be `NULL`, but at the time I assumed that `query` being `NULL` meant that there was a memory allocation failure. I think that [this](https://github.com/curl/curl/blob/44c9884552ad740fcec60d334f9601b205bf5b25/lib/ldap.c#L864) condition needs to also check if `conn->data->state.up.query` is not `NULL` and everything should just work? I won't have time to make this change and test it out for awhile, so someone else might need to do this.

**Comment by jay** [comment]
A partial fix for the out of memory error has landed in 8bb3a95. Now the error message is:
~~~
curl: (38) LDAP local: bind via ldap_win_bind Invalid Credentials
~~~

**Comment by jeroen** [comment]
This problem has also been reported in the R bindings. This image can be used to reproduce:

```
docker run --privileged -d -p 389:389 rroemhild/test-openldap
```

And then: `curl ldap://localhost:389`
